Xamarin.Forms.Platform.Android/AppCompat/NavigationPageRenderer.cs

```diff
@@ -829,9 +828,9 @@ void UpdateToolbar()
 			}
 			else
 			{
-				if (toggle != null)
+				if (toggle != null && _masterDetailPage != null)
```

Comment:
I'm wondering if we should just do the work to find the `MasterDetailPage` again here inside this `if` statement. I don't like the idea of holding a reference to a child page inside of a `NavigationPage`. Seems like we might get a memory leak at best and an `ObjectDisposedException` at worst if the hierarchy changes and that reference doesn't get updated soon enough.

Reply:
But we already do hold a reference of the renderer of `MasterDetailPage` in `_drawerLayout`. If we need to get a fresh reference of `MasterDetailPage`, we have to update `_drawerLayout` and `_drawerToggle`, which is made out of `_drawerlayout`, as well.
So basically, we don't get any advantage of doing the work in `UpdateToolbar` again without also updating `toggle`.